packages/ui5-application-inquirer/src/prompts/prompts.ts

```diff
@@ -0,0 +1,473 @@
+        name: promptNames.description,
+        guiOptions: {
+            hint: t('prompts.appDescTooltip'),
+            breadcrumb: true
+        },
+        message: t('prompts.appDescMessage'),
+        default: (answers: UI5ApplicationAnswers) => answers.description || t('prompts.appDescDefault')
+    } as InputQuestion<UI5ApplicationAnswers>;
+}
+
+/**
+ * Gets the `namespace` prompt.
+ *
+ * @param appName the application is used as part of namespace validation to determine max combined length
+ * @returns the `namespace` prompt
+ */
+function getNamespacePrompt(appName?: string): UI5ApplicationQuestion {
+    return {
+        type: 'input',
+        guiOptions: {
+            hint: t('prompts.appNamespaceTooltip'),
+            breadcrumb: true
+        },
+        name: promptNames.namespace,
+        message: t('prompts.appNamespaceMessage'),
+        default: (answers: UI5ApplicationAnswers) => answers.namespace ?? '',
+        validate: (namespace: string, answers: UI5ApplicationAnswers): boolean | string => {
+            if (namespace) {
+                return validateNamespace(namespace, answers.name || appName);
+            }
+            return true;
+        }
+    } as InputQuestion<UI5ApplicationAnswers>;
+}
+
+/**
+ * Gets the `title` prompt.
+ *
+ * @returns the `title` prompt
+ */
+function getTitlePrompt(): UI5ApplicationQuestion {
+    return {
+        type: 'input',
+        guiOptions: {
+            hint: t('prompts.appTitleTooltip'),
+            breadcrumb: true
+        },
+        name: promptNames.title,
+        message: t('prompts.appTitleMessage'),
+        default: (answers: UI5ApplicationAnswers) => answers.title || t('prompts.appTitleDefault')
+    } as InputQuestion<UI5ApplicationAnswers>;
+}
+
+/**
+ * Gets the `name` prompt.
+ *
+ * @param targetDir the directory path to search for exiting apps with the same name
+ * @param isCli if this prompt is to be executed on the CLI the validator will ensure the app name is unique for the target dir (usually `cwd`)
+ * @param isCapProject if the app is to be generated in a CAP project ensure that the name is unique within the CAP apps folder path
+ * @param appName the default app name, if not provided a default app name will be suggested
+ * @returns the UI5 application `name` prompt
+ */
+function getNamePrompt(
+    targetDir: string,
+    isCli: boolean,
+    isCapProject: boolean,
+    appName?: string
+): UI5ApplicationQuestion {
+    return {
+        type: 'input',
+        guiOptions: {
+            applyDefaultWhenDirty: true,
+            hint: t('prompts.appNameTooltip'),
+            mandatory: true,
+            breadcrumb: true
+        },
+        name: promptNames.name,
+        message: t('prompts.appNameMessage'),
+        default: (answers: UI5ApplicationAnswers) => answers.name || appName || defaultAppName(targetDir),
+        validate: (name, answers: UI5ApplicationAnswers): boolean | string => {
+            if (isCli || isCapProject) {
```

Comment:
Why would we call `validateAppName` if run via CLI and `validateModuleName` when running YUI?

Reply:
Good question, I had the same question actually. 

On the CLI you cannot update the target folder until after the app name has been input. However it is possible to provide the target folder and hide that prompt  (using prompt options or it uses `cwd` or in the case of CAP where the target folder is pre-determined) and so we need to check that this name is not an existing folder.

In YUI however the user will only get an error message under the target folder to indicate that the user must change the name or the target path. It is done this way as there is a limitation (I dont think YUI consider it a bug) that you cannot re-validate an earlier prompt on a step and so we would never get the error under the name prompt in YUI. 

This was investigated at time of implementation in App Gen and this was the only solution found. I did initially remove it for the move but found the same issue occurred so left it as is.  I can send you more detail internally.

Reply:
Thanks for changing it to `isYui` to better indicate that it is a required additional effort for the usage of that specific consumer.